### PR TITLE
Added a gateway example

### PR DESCRIPTION
**Pull request by galz10**

### CODE REVIEW

File: examples/complex/esp32/Gateway/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+## Configuration
+You need to configure the devices to connect over bluetooth in `ESP32_delegate.ino`.
+Make sure that the device_id you enter matches the device_id on the Google Cloud IoT core.
+The example configurations reflect models encountered during testing but you
+may need to add your own based on the data sheet for your device.
+
+Update the values in [ciotc_config.h](ciotc_config.h) so that they match the
+configuration for your gateway device and delegate devices.
+
+## Running the demo
```
Comment (gguuss) [inline examples/complex/esp32/Gateway/README.md:25]: Add instruction for adding devices in the console, maybe collapse the configuration and running sections into a numbered list:

1. Add devices in the console.
2. Update `ciotc_config.h` in Esp32-Gateway to match the devices.
3. Flash the gateway device with the project sources in Esp32-Gateway.
4. Flash the delegate device(s) with the project sources in Esp32-delegate.
5. Restart both devices...

File: examples/complex/esp32/Gateway/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        deviceid,state,get state;
+
+The gateway device will connect to the delegate over bluetooth and forward this command to the delegate device,
+then the gateway will receive the response and publish to the PubSub subscription.
+
+After you pull the message from the PubSub topic, you should then see the temperature read out or the state of the device depending on the command you sent.
+
+## Troubleshooting
+If the MQTT connection is resetting when you're attaching the delegate devices,
+make sure to check that you delegate devices are bounded to the gateway and that they are set up correctly.
```
Comment (gguuss) [inline examples/complex/esp32/Gateway/README.md:45]: replace:
make sure to check that you delegate devices are bounded to the gateway

with:
make sure that your delegate devices are bound to the gateway and that
the gateway is configured to use association-only in the [Cloud Console](https://console.cloud.google.com/iot).

File: library.properties [hunk 3]
```diff
@@ -1,4 +1,4 @@
-name=Google Cloud IoT Core JWT
+name=Google Cloud IoT Core JWT Gal
```
Comment (gguuss) [inline library.properties:1]: Let's leave the library properties as-is, these are used for publishing releases to Arduino for the library manager.

File: examples/complex/esp32/Gateway/Esp32-gateway/esp32-mqtt.h [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+// !!REPLACEME!!
+// The MQTT callback function for commands and configuration updates
+// Place your message handler code here.
+void messageReceived(String &topic, String &payload) {
+  int size = sizeof(delegate_device_id) / sizeof(delegate_device_id[0]);
+  Serial.println("incoming: " + topic + " - " + payload);
+  getDeviceID(payload);
+  incoming_payload = payload;
+
+  if(payload == "attach")
```
[the remark this answers is not in the document]
Comment (galz10) [reply examples/complex/esp32/Gateway/Esp32-gateway/esp32-mqtt.h:126]: So i changed the code a bit from how it was originally, so it doesn't automatically attach the devices , when the gateway connects you can send attach to the gateway and then it'll attach the devices in the array

**Comment by galz10** [comment]
@googlebot I fixed it.

File: examples/complex/esp32/Gateway/Esp32-gateway/esp32-mqtt.h [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+      private_key_str);
+
+  setupWifi();
+  netClient = new WiFiClientSecure();
+  mqttClient = new MQTTClient(360);
+  mqttClient->setOptions(180, true, 10000); // keepAlive, cleanSession, timeout
+  mqtt = new CloudIoTCoreMqtt(mqttClient, netClient, device);
+  mqtt->setUseLts(true);
+  mqtt->startMQTT();
+  mqttClient->subscribe("/devices/"+ String(device_id) +"/errors",1);
```
Comment (gguuss) [inline examples/complex/esp32/Gateway/Esp32-gateway/esp32-mqtt.h:230]: This should be QoS 0

**Pull request merged**